tests/algorithms/test_snapshot.py

```diff
@@ -320,37 +320,29 @@ def test_solve_wls(derived):
     assert isinstance(state_estimate,type(NavData()))
 
     # should have four rows
-    assert len(state_estimate.rows) == 4
+    assert len(state_estimate.rows) == 8
 
     # should have the following contents
+    assert "gps_millis" in state_estimate.rows
     assert "x_rx_m" in state_estimate.rows
     assert "y_rx_m" in state_estimate.rows
     assert "z_rx_m" in state_estimate.rows
     assert "b_rx_m" in state_estimate.rows
+    assert "lat_rx_deg" in state_estimate.rows
+    assert "lon_rx_deg" in state_estimate.rows
+    assert "alt_rx_deg" in state_estimate.rows
 
     # should have the same length as the number of unique timesteps
-    assert len(state_estimate) == len(np.unique(derived["gps_millis",:]))
+    assert len(state_estimate) == sum(1 for _ in derived.loop_time("gps_millis"))
```

Comment:
Nice!

Reply:
yeah, but I still think there should be a better groupby functionality so you don't have to iterate through all of loop_time just to get the group count number.